src/beman/exemplar/CMakeLists.txt

```diff
@@ -5,20 +5,21 @@ add_library(beman::exemplar ALIAS beman.exemplar)
 
 target_sources(beman.exemplar PRIVATE identity.cpp)
 
-target_include_directories(
-  beman.exemplar
-  PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../../../include>)
+set(INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../../include/)
+
+target_sources(
+  beman.exemplar PUBLIC
+  FILE_SET HEADERS
+  BASE_DIRS ${INCLUDE_DIR}
+  FILES ${INCLUDE_DIR}/beman/exemplar/identity.hpp)
+
+set_target_properties(beman.exemplar PROPERTIES VERIFY_INTERFACE_HEADER_SETS ON)
 
 install(
   TARGETS beman.exemplar
   EXPORT beman.exemplar
-  DESTINATION ${CMAKE_INSTALL_LIBDIR})
-
-install(
-  DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/../../../include/
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
```

Comment:
The CMake 3.24 docs seem to indicate we can leave off `DESTINATION` because FILE_SET with the TYPE HEADERS will implicitly be installed into the INCLUDEDIR DESTINATION:
https://cmake.org/cmake/help/v3.24/command/install.html

Reply:
CMake install faults when we remove this line for some reason.

Reply:
I will test again tomorrow morning with this line removed, but I am pretty sure when I run CMake 3.26 locally without the destination argument it faults at install?

Reply:
done!

Reply:
Ah I don't think removing this is correct. `cmake --install` no longer output the `identity.hpp` file, see latest CI run: https://github.com/beman-project/exemplar/actions/runs/11034273081/job/30647406013?pr=30#step:6:445 vs previous CI run: https://github.com/beman-project/exemplar/actions/runs/11024257637/job/30617095477#step:6:382 (Missing `#18 0.073 -- Installing: /opt/beman.exemplar/include/beman/exemplar/identity.hpp`)

Reply:
Reverting the change (restoring this line) produces the desired result, see: https://github.com/beman-project/exemplar/actions/runs/11034387505/job/30647795335?pr=30#step:6:389

Reply:
The problem is that `DESTINATION` is being applied to everything being installed. To make it apply only to the file set, it must come after `FILE_SET HEADERS`. I tried this and it worked on cmake 3.30.1.

From the [CMake docs](https://cmake.org/cmake/help/latest/command/install.html#signatures):

> The first ``<artifact-option>...`` group applies to target
>  :ref:`Output Artifacts` that do not have a dedicated group specified
>  later in the same call.

Reply:
👌 would update later.

Reply:
Oh, right. Good catch @camio. It helps to indent fields like `DESTINATION` so they are clearly "under" the `FILE_SET` it's modifying. Or you can split the `install()` command up into N more specific ones.

But I still swear that `FILE_SET` should imply `INCLUDEDIR` by default. Maybe @mathstuf can assist.

To be clear, I'm OK landing this PR with DESTINATION and following up on this detail later.

Reply:
Done, could @camio  or @bretbrownjr  provide more info on what this fixes? `cmake --install ...` still installs `gtest` aside from the project headers. Is this intended for having multiple source-file groups? Just curious.

Reply:
> Done, could @camio or @bretbrownjr provide more info on what [moving DESTINATION below FILE_SET] fixes?

It fixes the issue where `libbeman.exemplar.a` was being placed in `include/` instead of `lib/`.